src/main/java/org/folio/circulation/domain/RequestQueueRepository.java

```diff
@@ -95,4 +103,85 @@ public static RequestQueueRepository using(Clients clients) {
     }
     return requestUpdated.thenApply(r -> r.map(notUsed -> requestQueue));
   }
+
+  CompletableFuture<Result<RequestQueue>> reorderRequests(RequestQueue requestQueue) {
```

Comment:
Right, as far as I understand `updateRequestsWithChangedPositions` make a shift-like updates, i.e. if there are 3 requests in the queue and 2nd RQ is closed, than this method will handle this update as keep 1st RQ on the 1st position, move 3rd RQ to 2nd position as it is free now.

This won't work for our case. Because old requests positions and new positions are distributed randomly. In order to deal with this we have to:
1. Remove previous positions held by the changed requests.
1. Set the new positions.

That's what the method does.

Reply:
This seems particularly fragile. I wonder if we are approaching the point where we need to create an actual request queue in the DB and manage this there. I am concerned we could be updating a lot of requests and each could fail, leaving us in a weird state. Recovering from an error, for example after removing all positions, but before adding any new ones, seems challenging. Maybe it is not.

Either way, we should only have one request queue reorder method. Whether it is random order or numerical order shouldn't matter.

Reply:
Yep, unfortunately as is. It may cause issues...

I just think, can we think about batch update in circulation-storage? So, we will submit array of requests to update and they will be updated within the same transaction, so we will be able to properly handle it?

What do you think @mreno-EBSCO, @marcjohnson-kint?

Reply:
Hi @marcjohnson-kint, @mreno-EBSCO,

I've pushed experimental API https://github.com/folio-org/mod-circulation-storage/pull/213 as Marc advised. Please take a look if we can use it here.